Runtime/AvatarCreator/Scripts/AvatarTemplateFetcher.cs

```diff
@@ -36,19 +36,32 @@ public AvatarTemplateFetcher(CancellationToken ctx = default, TemplateVersions t
         /// Fetches all avatar templates without the icon renders via the avatarAPI.
         /// </summary>
         /// <returns></returns>
-        public async Task<List<AvatarTemplateData>> GetTemplates()
+        public async Task<List<AvatarTemplateData>> GetTemplates(OutfitGender gender = OutfitGender.None)
         {
             var templates = await avatarAPIRequests.GetAvatarTemplates();
-            switch (templateVersions)
+            return templates.Where(template => HasCorrectTemplateVersion(template) && HasCorrectGender(template)).ToList();
```

Comment:
With the gender filtering I was mostly thinking about being able to hide/show only male/female template avatar icons. I think it still makes sense to still download both male and female data + renders as it is very unlikely an app would only use 1 gender, so you may as well fetch both outfit genders at once initially and then in the SelectionElement you will then just set the visibility based on which gender you want to see. 

Also this now makes me thing that probably this is also the wrong place to be filtering the "Template version". Because the request for data (no renders) is super light weight and fast. So in that sense maybe it also makes sense to delegate the Template version filtering to the TemplateSelectionElement so that you just filter the ones you don't want before requesting the renders, if that makes sense.

Reply:
yep, agree with this.